
/* 
	Exception Handling :-
	_________________________________
	
	There are 2 ways to Handle Exceptions in methods .:
	1. Handle the exception there itself
	2. Allow it to propogate to be handled somewhere else.


	1. Handling the exception there itself .:
		We can use Try-Catch block for constructing an exception handler.
		If try block throws exception, then, catch block is used to catch and handle that exception.
		IF try does not throw any exception, then, catch block will get ignored.

	like .:

	try{
		write the code here.
	}
	catch(Exception1 e1){
		to handle exception1 ;
	}
	catch(Exception2 e2){
		to handle exception2 ;
	}
	finally{
		write the rest important executable codes here.
	}
*/

package PracticeFiles ;

import java.util.Scanner;

public class ExceptionHandlings{
	public static Scanner scan = new Scanner(System.in) ;
	
	public static void main(String [] args) {
		
		int numb = 0;
		//	suppose, we want the user to enter any number > 10.
		try {
			
			System.out.print("Enter a number ") ;
			numb = scan.nextInt() ;
			if(numb < 10) {
				throw new Exception("You are entering number less than 10.");
			}
			
		}
		
		catch(Exception e) {
			//	to get all the details of the exception : 
			System.out.println("The exception is " + e);
			System.out.println("Message is : " + e.getMessage()) ;
			System.out.println("Cause is : " + e.getCause());
			System.out.print("Stack Trace is : ");
			e.getStackTrace() ;
			System.out.println("Suppressed is : " + e.getSuppressed()) ;
			System.out.println("to string is : " + e.toString()) ;
		}
		
		finally {
			System.out.println("The input given by user is : " + numb);
		}
	}
}